### PR TITLE
Add GitHub Security Advisory Database 

### PR DESCRIPTION
**Pull request by masahiro331**
This Pull Request adds vulnerability information of github security advisory database to vuln-list.

The following Ecosystems are supported.
ref: https://developer.github.com/v4/enum/securityadvisoryecosystem/
- Composer
- Maven
- Npm
- Nuget
- Pip
- Rubygems

The data schema of trivy-db is undecided now.   
If you want to try of GraphQL API can be executed on this site.

https://developer.github.com/v4/explorer/

### Issues: 
- GitHub Security Advisory sometimes does not have CVE-ID
Therefore, it is stored using GHSA-ID.

### CODE REVIEW

File: ghsa/ghsa.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+			break
+		}
+
+		variables["cursor"] = githubql.NewString(getVulnerabilitiesQuery.PageInfo.EndCursor)
+	}
+
+	return ghsas, nil
+}
+
+func (c Config) saveGSA(dirName string, ghsaID, data interface{}) error {
```
Comment (simar7) [inline ghsa/ghsa.go]: Can we make `ghsaID` and `data` as their respective data types rather than an `interface` type?

File: ghsa/ghsa.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+		"total":     graphql.Int(100),
+		"cursor":    (*githubql.String)(nil),
+	}
+	for {
+		var err error
+		for i := 0; i <= c.Retry; i++ {
+			if i > 0 {
+				wait := math.Pow(float64(i), 2) + float64(utils.RandInt()%10)
+				log.Printf("retry after %f seconds\n", wait)
+				time.Sleep(time.Duration(time.Duration(wait) * time.Second))
```
Comment (simar7) [inline ghsa/ghsa.go]: This logic is currently untested, we could add a mock that fails on first try and passes subsequently.

File: ghsa/ghsa.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	bar.Finish()
+	return nil
+}
+
+func (c Config) FetchGithubSecurityAdvisories(ecosystem SecurityAdvisoryEcosystem) ([]GithubSecurityAdvisory, error) {
+	var getVulnerabilitiesQuery GetVulnerabilitiesQuery
+	var ghsas []GithubSecurityAdvisory
+	variables := map[string]interface{}{
+		"ecosystem": ecosystem,
+		"total":     graphql.Int(100),
```
Comment (simar7) [inline ghsa/ghsa.go]: Why is the count of `100` important? Can we increase that?

File: main.go [hunk 4]
```diff
@@ -146,6 +150,17 @@ func run() error {
 			return xerrors.Errorf("error in Photon update: %w", err)
 		}
 		commitMsg = "Photon Security Advisories"
+	case "ghsa":
+		src := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: githubToken},
```
Comment (simar7) [inline main.go]: I'm assuming this GitHub token will actually require the same level of permissions that a token to access the GraphQL api would?

File: ghsa/ghsa.go (hunk 1 above)
Comment (masahiro331) [reply ghsa/ghsa.go]: Sorry miss.

File: ghsa/ghsa.go (hunk 3 above)
Comment (masahiro331) [reply ghsa/ghsa.go]: The response size of security vulnerability is up to 100.

define const MaxResponseSize 100.

File: main.go (hunk 4 above)
Comment (masahiro331) [reply main.go]: graphql api does not require any scope...
Should I use another access token?

**Comment by masahiro331** [comment]
@simar7 
Thanks for the fix.
Consultation.
I think Ecosystem is better than Ecosystems for Config members.

For example,
FetchGithubSecurityAdvisories have  ecosystems of Config, but is designed to recieve the ecosystem as an argument.

**Pull request merged**